src/web/functions.php

```diff
@@ -9,9 +9,122 @@
  * @param  array  $airports
  * @return string[]
  */
-function getUniqueFirstLetters(array $airports)
+function getUniqueFirstLetters(array $airports): array
 {
-    // put your logic here
+    $firstLetters = [];
 
-    return ['A', 'B', 'C'];
-}
\ No newline at end of file
+    foreach ($airports as $airport) {
+        $firstLetters[] = $airport['name'][0];
+    }
+
+    sort($firstLetters);
+
+    return array_unique($firstLetters);
+}
+
+/**
+ * @param array $airports
+ * @param string $filterByFirstLetter
+ * @return array
+ */
+function filterByFirstLetter(array $airports, string $filterByFirstLetter): array
+{
+    $airportsByLetter = [];
+
+    foreach ($airports as $airport) {
+        if ($airport['name'][0] == $filterByFirstLetter) {
+            $airportsByLetter[] = $airport;
+        }
+    }
+
+    return $airportsByLetter;
+}
+
+/**
+ * @param array $airports
+ * @param string $filterByState
+ * @return array
+ */
+function filterByState(array $airports, string $filterByState): array
+{
+    $airportsByState = [];
+
+    foreach ($airports as $key => $airport) {
+        if ($airport['state'][0] === $filterByState) {
+            $airportsByState[] = $airport;
+        }
+    }
+
+    return $airportsByState;
+}
+
+/**
+ * @param array $airports
+ * @param string $sortAirports
+ * @return array
+ */
+function sortAirports(array $airports, string $sortAirports): array
+{
+    $airportsSort = [];
+
+    foreach ($airports as $key => $airport) {
+        $airportsSort[$key] = $airport[$sortAirports];
+    }
+
+    if (!empty($airports_sort)) {
+        array_multisort($airportsSort, SORT_ASC, $airports);
+    }
+
+    return $airports;
+}
+
+/**
+ * @param array $get
+ * @param array $link
+ * @return string
+ */
+function getLink(array $get, array $link = []): string
+{
+    $getParams = [];
+
+    if (isset($get['filter_by_first_letter'])) {
```

Comment:
Please put all these keys in an array and use foreach to iterate over all these keys. This will help avoid duplicating the` if (isset` logic.